crates/example-types/src/testable_delay.rs

```diff
@@ -0,0 +1,132 @@
+            max_time_in_milliseconds: 0,
+            fixed_time_in_milliseconds: 0,
+        }
+    }
+}
+
+#[derive(Eq, PartialEq, Debug, Clone, Default)]
+/// Settings for each type
+pub struct DelayConfig {
+    config: HashMap<SupportedTraitTypesForAsyncDelay, DelaySettings>,
+}
+
+impl DelayConfig {
+    pub fn new(config: HashMap<SupportedTraitTypesForAsyncDelay, DelaySettings>) -> Self {
+        DelayConfig { config }
+    }
+
+    pub fn add_settings_for_all_types(&mut self, settings: DelaySettings) {
+        let iterator = SupportedTraitTypesForAsyncDelayIterator::new();
+
+        for supported_type in iterator {
+            self.config.insert(supported_type, settings.clone());
+        }
+    }
+
+    pub fn add_setting(
+        &mut self,
+        supported_type: SupportedTraitTypesForAsyncDelay,
+        settings: &DelaySettings,
+    ) {
+        self.config.insert(supported_type, settings.clone());
+    }
+
+    pub fn get_setting(
+        &self,
+        supported_type: &SupportedTraitTypesForAsyncDelay,
+    ) -> Option<&DelaySettings> {
+        self.config.get(supported_type)
+    }
+}
+
+#[async_trait]
+/// Implement this method to add some delay to async call
+pub trait TestableDelay {
+    /// Add a delay from settings
+    async fn handle_async_delay(settings: &DelaySettings) {
+        match settings.delay_option {
+            DelayOptions::None => {}
+            DelayOptions::Fixed => {
+                async_sleep(Duration::from_millis(settings.fixed_time_in_milliseconds)).await;
+            }
+            DelayOptions::Random => {
+                let sleep_in_millis = rand::thread_rng().gen_range(
+                    settings.min_time_in_milliseconds..=settings.max_time_in_milliseconds,
+                );
+                async_sleep(Duration::from_millis(sleep_in_millis)).await;
+            }
+        }
+    }
+
+    /// Look for settings in the config and run it
+    async fn run_delay_settings_from_config(delay_config: &DelayConfig);
+}
+
+/// Iterator to iterate over enum
+struct SupportedTraitTypesForAsyncDelayIterator {
+    index: usize,
+}
+
+impl SupportedTraitTypesForAsyncDelayIterator {
+    fn new() -> Self {
+        SupportedTraitTypesForAsyncDelayIterator { index: 0 }
+    }
+}
+
+impl Iterator for SupportedTraitTypesForAsyncDelayIterator {
+    type Item = SupportedTraitTypesForAsyncDelay;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let supported_type = match self.index {
+            0 => Some(SupportedTraitTypesForAsyncDelay::Storage),
+            1 => Some(SupportedTraitTypesForAsyncDelay::ValidatedState),
+            2 => Some(SupportedTraitTypesForAsyncDelay::BlockHeader),
+            _ => {
+                assert_eq!(self.index, 3, "Need to ensure that newly added or removed `SupportedTraitTypesForAsyncDelay` enum is handled in iterator");
+                return None;
+            }
+        };
+        self.index += 1;
+        supported_type
+    }
+}
```

Comment:
What is the point of implementing iterator?

Reply:
add the same settings for each type

Reply:
basically so you dont have to do:
config.add_setting(SupportedTraitTypesForAsyncDelay::Storage, &delay_settings);
config.add_setting(SupportedTraitTypesForAsyncDelay::BlockHeader, &delay_settings);
...